lib/ansible/modules/packaging/os/apt_key.py

```diff
@@ -211,11 +229,19 @@ def download_key(module, url):
         module.fail_json(msg="error getting key id from url: %s" % url, traceback=format_exc())
 
 
-def import_key(module, keyring, keyserver, key_id):
+def import_key(module, keyring, keyserver, keyserver_options, key_id):
+    if (environ.get('http_proxy')):
```

Comment:
If you're supporting http_proxy, you should also support no_proxy. Is https_proxy also relevant here? Seems like it would be if a key server could ever be https.

Reply:
I actually have both https_proxy and no_proxy in the code, had to took them out pretty quickly, because they are not supported options, see error message below if you pass them in from --keyserver-options:
```
sudo apt-key adv --keyserver-options "http-proxy=http://username:password@proxy.example.com:8080 https-proxy=http://username:password@proxy.example.com:8080 https_proxy=http://username:password@proxy.example.com:8080 no-proxy=www.google.com no_proxy=www.google.com" --keyserver keyserver.ubuntu.com --recv-keys GPG_KEY
Executing: /tmp/apt-key-gpghome.zJ3uDH1oNy/gpg.1.sh --keyserver-options http-proxy=http://username:password@proxy.example.com:8080 https-proxy=http://username:password@proxy.example.com:8080 https_proxy=http://username:password@proxy.example.com:8080 no-proxy=www.google.com no_proxy=www.google.com --keyserver keyserver.ubuntu.com --recv-keys GPG_KEY
gpg: keyserver option 'https-proxy' is unknown
gpg: keyserver option 'https_proxy' is unknown
gpg: keyserver option 'no-proxy' is unknown
gpg: keyserver option 'no_proxy' is unknown
gpg: "GPG_KEY" not a key ID: skipping

```
below are my speculation due to the lack of documentation on keyserver-options:
* no proxy, this apt-key is hitting single URL(the key server) which is either in the proxy or outside the proxy, so no_proxy or such is needed. I guess apt-key will fail when there is a URL forward or redirect on key server that cross proxy (not sure if it's allowed by OpenPGP)
* https proxy, according to the answers on this: https://security.stackexchange.com/questions/4161/shouldnt-gpg-key-fetching-use-a-secure-connection
it's always on http protocal so https_proxy is not releveant and not supported.

Reply:
In that case, properly supporting no_proxy would require parsing whether any domain suffix in no_proxy is a suffix of the keyserver requested.  If it is, the http-proxy option should not be passed.

Reply:
Hi @jamescassell 
I have implemented the suggested logic